### PR TITLE
Feature/44213 spot switch with field

### PR DESCRIPTION
**Pull request by b12f**
Closes https://community.openproject.org/work_packages/44213/activity

### CODE REVIEW

File: frontend/src/app/spot/components/switch/switch.component.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  Output,
+  ViewChild,
+} from '@angular/core';
+import {
+  ControlValueAccessor,
+  NG_VALUE_ACCESSOR,
+} from '@angular/forms';
+import { populateInputsFromDataset } from 'core-app/shared/components/dataset-inputs';
+
+export type SpotSwitchState = boolean;
```
Comment (oliverguenther) [inline frontend/src/app/spot/components/switch/switch.component.ts:19]: Why add a new type for boolean? I don't expect the type to change, so I'd propose to keep it at boolean

File: frontend/src/app/spot/components/switch/switch.component.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  writeValue(value:SpotSwitchState):void {
+    // This is set in a timeout because the initial value is set before the template is ready,
+    // which causes the input nativeElement to not be available yet.
+    setTimeout(() => {
+      const input = this.input.nativeElement as HTMLInputElement;
+      if (value === null) {
+        input.indeterminate = true;
+      } else {
+        input.indeterminate = false;
+      }
```
Comment (oliverguenther) [inline frontend/src/app/spot/components/switch/switch.component.ts]:
```suggestion
      input.indeterminate = value === null;
```

File: frontend/src/app/spot/components/switch/switch.component.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+      this.checked = !!value;
```
Comment (oliverguenther) [inline frontend/src/app/spot/components/switch/switch.component.ts]: Is this timeout necessary if we just use a ngModel similar to how the toggle works? In this case, writeValue would be simpler

**Review by oliverguenther — CHANGES_REQUESTED** [review]
Some minor remarks and `onPush` is missing which is breaking using it in the datepicker right now

File: frontend/src/app/spot/components/switch/switch.component.ts (hunk 1 above)
Comment (b12f) [reply frontend/src/app/spot/components/switch/switch.component.ts:19]: In a sense a lazy copy/paste from spot-checkbox, but I thought I actually kinda liked having the interfacing value type represented near the top of the file.

Comment (oliverguenther) [reply frontend/src/app/spot/components/switch/switch.component.ts:19]: :+1:  fine for me if we keep this practice. I just found it weird to see a custom type mapped to a boolean.

**Review by oliverguenther — APPROVED** [review]

**Pull request merged**